Sources/IO/Geometry/IFCImporter/index.js

```diff
@@ -244,32 +245,40 @@ function vtkIFCImporter(publicAPI, model) {
   publicAPI.importActors = (renderer) => {
     if (model.mergeGeometries) {
       const opaqueMeshes = meshes.filter((mesh) => mesh.color.w === 1);
-      let apd = vtkAppendPolyData.newInstance();
+      const oapd = vtkAppendPolyData.newInstance();
 
       opaqueMeshes.forEach((mesh) => {
         const pd = createColoredPolyDataFromIFCMesh(mesh);
-        apd.addInputData(pd);
+        oapd.addInputData(pd);
       });
 
+      const opdn = vtkPolyDataNormals.newInstance();
```

Comment:
is this overwritting existing normals ? If so, it might not be ideal. 
vtkPolyDataNormals does not perform well in case of sharp edges.

Reply:
this is only for "merged geometries" which create two actors one to group opaque meshes and one transparent to group transparent meshes, the default option will create actors with their correct normals.

Reply:
Would you say that the AppendPolyData is "dropping" the normals of the input polydatas ?

Reply:
By looking at the code, it doesn't seems to be the case

Reply:
vtkAppendPolyData seems to preserve normals if they exist in ALL the input polydatas.

So why do you need to recompute normals here ? Is is because some polydata are missing normals ?

Reply:
i think it should be fixed